runtime_scan/pkg/provider/aws/config.go

```diff
@@ -15,40 +15,89 @@
 
 package aws
 
-import "github.com/spf13/viper"
+import (
+	"fmt"
+
+	"github.com/spf13/viper"
+)
 
 const (
-	AWSSubnetID            = "AWS_SUBNET_ID"
-	AWSJobImageID          = "AWS_JOB_IMAGE_ID"
-	AWSSecurityGroupID     = "AWS_SECURITY_GROUP_ID"
-	AWSInstanceType        = "AWS_INSTANCE_TYPE"
-	defaultAWSJobImageID   = "ami-0568773882d492fc8" // ubuntu server 22.04 LTS (HVM), SSD volume type
-	defaultAWSInstanceType = "t2.large"
+	EnvScannerRegion       = "VMCLARITY_AWS_SCANNER_REGION"
+	EnvSubnetID            = "VMCLARITY_AWS_SUBNET_ID"
+	EnvSecurityGroupID     = "VMCLARITY_AWS_SECURITY_GROUP_ID"
+	EnvKeyPairName         = "VMCLARITY_AWS_KEYPAIR_NAME"
+	EnvScannerImage        = "VMCLARITY_AWS_SCANNER_AMI_ID"
+	EnvScannerInstanceType = "VMCLARITY_AWS_SCANNER_INSTANCE_TYPE"
+	EnvBlockDeviceName     = "VMCLARITY_AWS_BLOCK_DEVICE_NAME"
+
+	DefaultScannerInstanceType = "t2.large"
+	DefaultBlockDeviceName     = "xvdh"
 )
 
 type Config struct {
-	AmiID           string // image id of a scanner job
-	SubnetID        string // the scanner's subnet ID
-	SecurityGroupID string // the scanner's security group
-	InstanceType    string // the scanner's instance type
+	// Region where the Scanner instance needs to be created
+	ScannerRegion string
+	// SubnetID where the Scanner instance needs to be created
+	SubnetID string
+	// SecurityGroupID which needs to be attached to the Scanner instance
+	SecurityGroupID string
+	// KeyPairName is the name of the SSH KeyPair to use for Scanner instance launch
+	KeyPairName string
+	// ScannerImage is the AMI image used for creating Scanner instance
+	ScannerImage string
+	// ScannerInstanceType is the instance type used for Scanner instance
+	ScannerInstanceType string
+	// BlockDeviceName contains the block device name used for attaching Scanner volume to the Scanner instance
+	BlockDeviceName string
 }
 
-func setConfigDefaults() {
-	viper.SetDefault(AWSJobImageID, defaultAWSJobImageID)
-	viper.SetDefault(AWSInstanceType, defaultAWSInstanceType)
+func (c *Config) Validate() error {
+	if c.ScannerRegion == "" {
+		return fmt.Errorf("parameter Region must not be nil")
```

Comment:
I think "must be provided" would be better than "must not be nil" so as to not leak through go language?

Reply:
Alright. Will update the wording here.

Reply:
Fixed.